### DISCUSSION TITLE
画布放大缩小或拖拽后，如何使新增的节点（拖拽节点到画布）还在鼠标松开的位置

### DISCUSSION THREAD

**Discussion by fengxuyan** [antvis/X6#2499] [Q&A]
在不使用stencil和dnd的情况下，使用原生drag事件拖拽一个自定义的div，在drop到画布时新增一个节点，新增节点位置用的是pageX和pageY。但是在画布拖拽或者缩放后，此时再用pageX和pageY作为节点位置就不对了，需要得到一个相对于画布的位置。如何在drop事件发生后，获取到鼠标松开时相对于画布的位置坐标呢

**Comment by ycx0303** [comment]
+1

**Comment by ycx0303** [comment]
> 在不使用stencil和dnd的情况下，使用原生drag事件拖拽一个自定义的div，在drop到画布时新增一个节点，新增节点位置用的是pageX和pageY。但是在画布拖拽或者缩放后，此时再用pageX和pageY作为节点位置就不对了，需要得到一个相对于画布的位置。如何在drop事件发生后，获取到鼠标松开时相对于画布的位置坐标呢

请问有解决吗？

**Comment by ycx0303** [comment]
解决了
```
const { x, y } = graph.clientToLocal(e.clientX, e.clientY)；
graph.addNode({
  x,
  y
})
```
